### PR TITLE
Add note about timing of type loader exceptions

### PR DESCRIPTION
**Pull request by jkotas**
Describe your changes here.

Add note about timing of type loader exceptions based on the feedback received offline.

Also delete notes about small behavior differences with missing PathDiscoveryPermissions. PathDiscoveryPermissions is part of Code Access Security that was deprecated long time ago. No need to compilate the docs for it.

### CODE REVIEW

File: xml/System.IO/FileLoadException.xml [hunk 1]
```diff
@@ -59,15 +60,15 @@
   
  `FileLoadException` has the default HRESULT of COR_E_FILELOAD, which has the value 0x80131621, but this is not the only possible HRESULT.  
   
- If your code does not have <xref:System.Security.Permissions.FileIOPermissionAttribute.PathDiscovery%2A> permission, the error message for this exception may only contain file or directory names instead of fully qualified paths.  
+ The exact timing of when the static assembly references are loaded is unspecified. This exception may be thrown before the method that is statically referencing the assembly that failed to load starts executing.
```
Comment (gewarren) [inline xml/System.IO/FileLoadException.xml:63]:
```suggestion
The exact timing of when static assembly references are loaded is unspecified. This exception may be thrown before the method that statically references the assembly that failed to load starts executing.
```

File: xml/System/MissingMethodException.xml [hunk 2]
```diff
@@ -64,7 +64,9 @@
  <xref:System.MissingMethodException> uses the HRESULT COR_E_MISSINGMETHOD, that has the value 0x80131513.  
   
  For a list of initial property values for an instance of <xref:System.MissingMethodException>, see the <xref:System.MissingMethodException.%23ctor%2A> constructors.  
-  
+ 
+ The exact timing of when the statically referenced methods are loaded is unspecified. This exception may be thrown before the method that is referencing the missing method starts executing.
```
Comment (gewarren) [inline xml/System/MissingMethodException.xml:68]:
```suggestion
The exact timing of when statically referenced methods are loaded is unspecified. This exception may be thrown before the method that references the missing method starts executing.
```

File: xml/System/TypeLoadException.xml [hunk 3]
```diff
@@ -71,6 +71,8 @@
   
  For a list of initial property values for an instance of <xref:System.TypeLoadException>, see the <xref:System.TypeLoadException.%23ctor%2A> constructors.  
   
+ The exact timing of when the statically references types are loaded is unspecified. This exception may be thrown before the method that is referencing the missing type starts executing.
```
Comment (gewarren) [inline xml/System/TypeLoadException.xml:74]:
```suggestion
The exact timing of when statically referenced types are loaded is unspecified. This exception may be thrown before the method that references the missing type starts executing.
```

**Review by gewarren — APPROVED** [review]
LGTM

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you for improving the docs @jkotas !

**Pull request merged**